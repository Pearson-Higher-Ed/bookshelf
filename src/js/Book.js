import React, { Component } from 'react';
import Dialog from 'material-ui/Dialog';
import SvgIcon from 'material-ui/SvgIcon';

export default class Book extends Component {
  constructor(props) {
    super(props);

    this.handleBookClick = this.handleBookClick.bind(this);
    this.handleModalOpen = this.handleModalOpen.bind(this);
    this.handleModalClose = this.handleModalClose.bind(this);
    
    this.state = {
      modalOpen: false
    };
  }

  handleBookClick() {
    if (this.props.onBookClick) {
      this.props.onBookClick(this.props.id);
    }
    else {
      location.href='book.html?bookId=' + this.props.id;
    }    
  }
  
  handleModalOpen() {
    this.setState({modalOpen: true});
  }
  
  handleModalClose() {
    this.setState({modalOpen: false});
  }

  renderImage(bookCoverExists) {
    if (bookCoverExists) {
      return(
        <img className="image" src={this.props.image} />
      )
    }
    else {
      return (<span className="image"></span>)
    }
  }

  render() {
    const bookCoverExists = (this.props.image !== '');
    const style = {
      modal:{
        width:'59%',
        maxWidth: '59%',
        bodyStyle: {
          padding:'5.9% 6.4%'
        }
      },
      cancelIcon: {
        color: '#8d8d8d',
        position: 'absolute',
        top: '19px',
        right: '19px',
        height: '18.7px',
        width: '18px',
        cursor: 'pointer'
      },
      moreInfo: {
        height: '18px',
        width: '18px'
      }    
    }
    const MoreinfoIcon = (props) => (
      <SvgIcon {...props}>
        <path d="M192.984304,258.45713 L189.015304,258.45713 C188.559435,258.45713 188.188478,258.086565 188.188478,257.630696 C188.188478,257.174826 188.559435,256.803478 189.015304,256.803478 L190.172783,256.803478 L190.172783,252.503826 L189.677,252.503826 C189.22113,252.503826 188.850174,252.133261 188.850174,251.677391 C188.850174,251.221522 189.22113,250.850565 189.677,250.850565 L191,250.850565 C191.45587,250.850565 191.826826,251.221522 191.826826,251.677391 L191.826826,256.803478 L192.984304,256.803478 C193.440565,256.803478 193.811522,257.174826 193.811522,257.630696 C193.811522,258.086565 193.440565,258.45713 192.984304,258.45713 M190.668957,247.54287 C191.307565,247.54287 191.826826,248.062522 191.826826,248.700348 C191.826826,249.338565 191.307565,249.857826 190.668957,249.857826 C190.030739,249.857826 189.511478,249.338565 189.511478,248.700348 C189.511478,248.062522 190.030739,247.54287 190.668957,247.54287 M191,244 C186.037478,244 182,248.037478 182,253 C182,257.962913 186.037478,262 191,262 C195.962522,262 200,257.962913 200,253 C200,248.037478 195.962522,244 191,244" id="ic_info" fill="#b7bbbf"></path>
      </SvgIcon>
    );
    const CancelIcon = (props) => (
      <SvgIcon {...props}>
        <path d="M712.993036,23.3253012 L720.736289,15.2808193 C721.026024,14.9878313 721.026024,14.5128916 720.735855,14.2199036 C720.445687,13.9266988 719.975518,13.9266988 719.685349,14.2199036 L711.976795,22.005012 L704.268458,14.2199036 C703.978072,13.9266988 703.50812,13.9266988 703.217735,14.2199036 C702.927566,14.5128916 702.927566,14.9878313 703.217518,15.2808193 L710.960554,23.3253012 L703.217735,31.3697831 C702.927566,31.6627711 702.927566,32.1377108 703.217735,32.4306988 C703.507687,32.7234699 703.978072,32.7239036 704.268458,32.4306988 L711.976795,24.6455904 L719.685349,32.4306988 C719.975518,32.7239036 720.44612,32.7234699 720.735855,32.4306988 C721.026024,32.1377108 721.026024,31.6627711 720.735855,31.3697831 L712.993036,23.3253012 Z" id="ic_cancel" fill="#8d8d8d"></path>
      </SvgIcon>
    );
    return (
      <div className={`book ${bookCoverExists ? '' : 'no-book-cover'}`}>
        <a href="javascript:void(0);"
           className="container"
           onClick={this.handleBookClick}
           tabIndex="0">
          {this.renderImage(bookCoverExists)}
          <p className="title">{this.props.title}</p>
        </a>        
        <a className="info"
           href="javascript:void(0);"
           onClick={this.handleModalOpen}
           tabIndex="0">
          <MoreinfoIcon viewBox="182 244 18 18" style={style.moreInfo}/>
        </a>
        <Dialog
          modal={false}
          open={this.state.modalOpen}
          onRequestClose={this.handleModalClose}
          className="modal"
          contentStyle={style.modal}
          bodyStyle= {style.modal.bodyStyle}>
          <CancelIcon onClick={this.handleModalClose} viewBox="703 14 18 18.7" style={style.cancelIcon}/>            
          <div className={`image-container ${bookCoverExists ? '' : 'no-book-cover'}`}>
            {this.renderImage(bookCoverExists)}
          </div>
          <div className="meta-container">
            <p className="header">PUBLICATION INFO</p>
            <p className="body">{this.props.title}</p>
            <p className="footer">{this.props.author}</p>
            <p className="header course-info">COURSE INFO</p>
            <p className="body">Course Name</p>
            <p className="footer">CourseId123</p>
          </div>
          <div className="desc-container">
            {this.props.description}
          </div>                
        </Dialog>
      </div>
    )
  }
}
