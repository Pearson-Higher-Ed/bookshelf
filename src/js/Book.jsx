/**
PEARSON PROPRIETARY AND CONFIDENTIAL INFORMATION SUBJECT TO NDA
 *  Copyright Â© 2017 Pearson Education, Inc.
 *  All Rights Reserved.
 *
 * NOTICE:  All information contained herein is, and remains
 * the property of Pearson Education, Inc.  The intellectual and technical concepts contained
 * herein are proprietary to Pearson Education, Inc. and may be covered by U.S. and Foreign Patents,
 * patent applications, and are protected by trade secret or copyright law.
 * Dissemination of this information, reproduction of this material, and copying or distribution of this software
 * is strictly forbidden unless prior written permission is obtained
 * from Pearson Education, Inc.
* */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Dialog from 'material-ui/Dialog';
import { messages } from './defaultMessages';

export default class Book extends Component {
  constructor(props) {
    super(props);

    this.handleBookClick = this.handleBookClick.bind(this);
    this.handleModalOpen = this.handleModalOpen.bind(this);
    this.handleModalClose = this.handleModalClose.bind(this);

    this.state = {
      modalOpen: false
    };
  }

  handleBookClick(e) {
    if (e && e.preventDefault) {
      e.preventDefault();
    }
    if (this.props.onBookClick) {
      if (this.props.book.type) {
        this.props.onBookClick(this.props.id, this.props.book.type);
        this.props.storeBookDetails(this.props.book);
      } else {
        this.props.onBookClick(this.props.id);
      }
    } else {
      window.location.href = `book.html?bookId=${this.props.id}`;
    }
  }

  handleModalOpen(e) {
    if (e && e.preventDefault) {
      e.preventDefault();
    }
    this.setState({ modalOpen: true });
  }

  handleModalClose(e) {
    if (e && e.preventDefault) {
      e.preventDefault();
    }
    this.setState({ modalOpen: false });
  }

  renderImage(bookCoverExists) {
    if (bookCoverExists) {
      return (
        <img className="image" src={this.props.image} alt={`${this.props.title} cover`} />
      );
    }

    return (<span className="image" />);
  }

  render() {
    const { formatMessage } = this.props.intl;
    const bookCoverExists = (this.props.image !== '');
    const style = {
      modal: {
        width: '59%',
        maxWidth: '59%',
        bodyStyle: {
          padding: '5.9% 6.4%'
        }
      }
    };

    return (
      <div className={`book ${bookCoverExists ? '' : 'no-book-cover'}`}>
        <a
          className="bookContainer"
          onClick={this.handleBookClick}
          onKeyPress={this.handleBookClick}
          role="link"
          tabIndex="0"
        >
          {this.renderImage(bookCoverExists)}
          <div
            className="title"
            onClick={this.handleBookClick}
            onKeyPress={this.handleBookClick}
            aria-label={this.props.title}
            role="link"
            tabIndex="0"
          >
            {this.props.title}
          </div>
        </a>
        <a
          className="info"
          role="link"
          onClick={this.handleModalOpen}
          onKeyPress={this.handleModalOpen}
          tabIndex="0"
          aria-label={formatMessage(messages.moreInfo)}
        >{''}</a>
        <Dialog
          modal={false}
          open={this.state.modalOpen}
          onRequestClose={this.handleModalClose}
          className="modal"
          contentStyle={style.modal}
          bodyStyle={style.modal.bodyStyle}
        >
          <div
            className="cancelBtn"
            role="presentation"
            onClick={this.handleModalClose}
            aria-label={formatMessage(messages.moreInfoCloseIcon)}
          />
          <div className={`image-container ${bookCoverExists ? '' : 'no-book-cover'}`}>
            {this.renderImage(bookCoverExists)}
          </div>
          <div className="meta-container">
            <p className="header">{formatMessage(messages.publicationInfo)}</p>
            <p className="body">{this.props.title}</p>
            <p className="footer">{this.props.author}</p>
            <p className="header course-info">{formatMessage(messages.courseInfo)}</p>
            <p className="body">Course Name</p>
            <p className="footer">CourseId123</p>
          </div>
          <div className="desc-container">
            {this.props.description}
          </div>
        </Dialog>
      </div>
    );
  }
}

Book.defaultProps = {
  storeBookDetails: null
};

Book.propTypes = {
  description: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  onBookClick: PropTypes.func.isRequired,
  book: PropTypes.object.isRequired,
  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  intl: PropTypes.object.isRequired,
  storeBookDetails: PropTypes.func
};
