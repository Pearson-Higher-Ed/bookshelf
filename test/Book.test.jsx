/**
PEARSON PROPRIETARY AND CONFIDENTIAL INFORMATION SUBJECT TO NDA
 *  Copyright Â© 2017 Pearson Education, Inc.
 *  All Rights Reserved.
 *
 * NOTICE:  All information contained herein is, and remains
 * the property of Pearson Education, Inc.  The intellectual and technical concepts contained
 * herein are proprietary to Pearson Education, Inc. and may be covered by U.S. and Foreign Patents,
 * patent applications, and are protected by trade secret or copyright law.
 * Dissemination of this information, reproduction of this material, and copying or distribution of this software
 * is strictly forbidden unless prior written permission is obtained
 * from Pearson Education, Inc.
**/

/* eslint-disable */

import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider, injectIntl } from 'react-intl';
import injectTapEventPlugin from 'react-tap-event-plugin';
import { InternationalSupport } from '@pearson-incubator/aquila-js-core';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'
import Dialog from 'material-ui/Dialog';
import Book from '../src/js/Book';
import msgObject from '../translations';

const intlObj = new InternationalSupport(msgObject, 'en');
injectTapEventPlugin();

const bookData = {
  iseT1: false
}
const bookdataeT1 = {
  iseT1: true
}
const BookComp = injectIntl(({date, intl}) => (
    <Book key='123'
    		intl={intl}
        id='123'
        author='author'
        title='title'
        image='image'
        description='description'
        onBookClick={function(){}}
        book={bookData}
        ></Book>
));
const BookCompeT1 = injectIntl(({date, intl}) => (
    <Book key='123'
        intl={intl}
        id='123'
        author='author'
        title='title'
        image='image'
        description='description'
        onBookClick={function(){}}
        storeBookDetails={function(){}}
        book={bookdataeT1}
        />
));

/*const BookCompWithoutBookClk = injectIntl(({date, intl}) => (
    <Book key='123'
        intl={intl}
        id='123'
        author='author'
        title='title'
        image='image'
        description='description'
        storeBookDetails={function(){}}
        book={bookData}
        />
));*/


it('renders correctly', () => {
  const tree = renderer
    .create(<MuiThemeProvider><IntlProvider locale={intlObj.getLocale()} messages={intlObj.getMessages()}><BookComp /></IntlProvider></MuiThemeProvider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});


it('renders the Book', () => {
  	const component = renderer.create(<MuiThemeProvider><IntlProvider locale={intlObj.getLocale()} messages={intlObj.getMessages()}><BookComp /></IntlProvider></MuiThemeProvider>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  tree.children[0].props.onClick({preventDefault(){}});
  tree.children[1].props.onClick({preventDefault(){}});
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders eT1 Book', () => {
    const component = renderer.create(<MuiThemeProvider><IntlProvider locale={intlObj.getLocale()} messages={intlObj.getMessages()}><BookCompeT1 /></IntlProvider></MuiThemeProvider>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  tree.children[0].props.onClick({preventDefault(){}});
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

/*it('renders without book click', () => {
    const component = renderer.create(<MuiThemeProvider><IntlProvider locale={intlObj.getLocale()} messages={intlObj.getMessages()}><BookCompWithoutBookClk /></IntlProvider></MuiThemeProvider>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  tree.children[0].props.onClick({preventDefault(){}});
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});*/

