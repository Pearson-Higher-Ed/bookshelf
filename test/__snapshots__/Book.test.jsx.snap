// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="book "
>
  <a
    className="bookContainer"
    onClick={[Function]}
    role="link"
    tabIndex="0"
  >
    <img
      alt="title cover"
      className="image"
      src="image"
    />
    <p
      className="title"
    >
      title
    </p>
  </a>
  <a
    aria-label="More Info"
    className="info"
    onClick={[Function]}
    role="link"
    tabIndex="0"
  >
    
  </a>
</div>
`;

exports[`renders eT1 Book 1`] = `
<div
  className="book "
>
  <a
    className="bookContainer"
    onClick={[Function]}
    role="link"
    tabIndex="0"
  >
    <img
      alt="title cover"
      className="image"
      src="image"
    />
    <p
      className="title"
    >
      title
    </p>
  </a>
  <a
    aria-label="More Info"
    className="info"
    onClick={[Function]}
    role="link"
    tabIndex="0"
  >
    
  </a>
</div>
`;

exports[`renders eT1 Book 2`] = `
<div
  className="book "
>
  <a
    className="bookContainer"
    onClick={[Function]}
    role="link"
    tabIndex="0"
  >
    <img
      alt="title cover"
      className="image"
      src="image"
    />
    <p
      className="title"
    >
      title
    </p>
  </a>
  <a
    aria-label="More Info"
    className="info"
    onClick={[Function]}
    role="link"
    tabIndex="0"
  >
    
  </a>
</div>
`;

exports[`renders the Book 1`] = `
<div
  className="book "
>
  <a
    className="bookContainer"
    onClick={[Function]}
    role="link"
    tabIndex="0"
  >
    <img
      alt="title cover"
      className="image"
      src="image"
    />
    <p
      className="title"
    >
      title
    </p>
  </a>
  <a
    aria-label="More Info"
    className="info"
    onClick={[Function]}
    role="link"
    tabIndex="0"
  >
    
  </a>
</div>
`;

exports[`renders the Book 2`] = `
<div
  className="book "
>
  <a
    className="bookContainer"
    onClick={[Function]}
    role="link"
    tabIndex="0"
  >
    <img
      alt="title cover"
      className="image"
      src="image"
    />
    <p
      className="title"
    >
      title
    </p>
  </a>
  <a
    aria-label="More Info"
    className="info"
    onClick={[Function]}
    role="link"
    tabIndex="0"
  >
    
  </a>
</div>
`;
